kratos/solving_strategies/convergencecriterias/residual_criteria.h

```diff
@@ -325,6 +326,44 @@ class ResidualCriteria : public  ConvergenceCriteria< TSparseSpace, TDenseSpace
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief This method computes the norm of the residual
+     * @details It checks if the dof is fixed
+     * @param rResidualSolutionNorm The norm of the residual
+     * @param rDofNum The number of DoFs
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param b RHS vector (residual + reactions)
+     */
+    void CalculateResidualNorm(
+        TDataType& rResidualSolutionNorm,
+        SizeType& rDofNum,
+        DofsArrayType& rDofSet,
+        const TSystemVectorType& b
+        )
+    {
+        // Initialize
+        TDataType residual_solution_norm = TDataType();
+        SizeType dof_num = 0;
+
+        // Loop over Dofs
+        #pragma omp parallel for reduction(+:residual_solution_norm,dof_num)
+        for (int i = 0; i < static_cast<int>(rDofSet.size()); i++) {
+            auto it_dof = rDofSet.begin() + i;
+
+            IndexType dof_id;
+            TDataType residual_dof_value;
+
+            if (it_dof->IsFree()) {
```

Comment:
while this one IS important, since the residual will be nonzero even at convergence when the dof is fixed